woocommerce/woocommerce-admin
# Add Stripe & PayPal Connection Flows

Part of #2684. 

This PR adds the Stripe & PayPal connection flows to the payments task list. 

PayPal does not require WCS for the OAuth flow, so two modes are supported here. OAuth connection, or manual key entry if the OAuth connection fails.

Stripe does require WCS. There is an auto-create account mode, a Oauth flow, and manual configuration.

<img width="747" alt="Screen Shot 2019-09-04 at 4 33 03 PM" src="https://user-images.githubusercontent.com/689165/64290472-29520180-cf34-11e9-855f-f64b75dc713b.png">

<img width="731" alt="Screen Shot 2019-09-04 at 4 33 25 PM" src="https://user-images.githubusercontent.com/689165/64290491-32db6980-cf34-11e9-86bc-059f321bf79f.png">

<img width="706" alt="Screen Shot 2019-09-04 at 4 51 25 PM" src="https://user-images.githubusercontent.com/689165/64290533-3ff85880-cf34-11e9-9cd4-c0a59b04982f.png">

### Screenshots

<img width="737" alt="Screen Shot 2019-09-05 at 3 31 13 PM" src="https://user-images.githubusercontent.com/689165/64374664-5238cc00-cff2-11e9-9426-3dc4239dcab5.png">

<img width="722" alt="Screen Shot 2019-09-05 at 3 31 54 PM" src="https://user-images.githubusercontent.com/689165/64374675-57961680-cff2-11e9-8ee4-8cf23c3fb43c.png">

### Detailed test instructions:

* Delete the `woocommerce_onboarding_payments` option if you have previously tested this branch, the Stripe branch, or want to run through the steps again. We will clear this automatically in the future.

PayPal:

* Go under your payment settings and make sure `API Username` and `API Password` are blank for PayPal Checkout.
* Go to the payments task, and make sure PayPal is selected. Attempt to connect through the OAuth flow. You will need a PayPal Business account.
* Test the manual mode by clearing the above onboarding option, and trying again. Instead of clicking the connect button, manually set your URL to `/wp-admin/admin.php?page=wc-admin&task=payments&paypal-connect=0`.

Stripe:

* Go under your payment settings and make sure `Live Publishable Key` and `Live Secret Key` are blank for Stripe.

See testing directions on https://github.com/woocommerce/woocommerce-admin/pull/2887.
